C++/C++.sublime-syntax

```diff
@@ -608,13 +618,19 @@ contexts:
       captures:
         1: keyword.control.module.c++
       push:
+        - match: '\.'
+          scope: invalid.illegal.unexpected-character.c++
+          push:
+            - match: '.*$'
+              pop: true
```

Comment:
Same here, but for module names.

```suggestion
        - match: (\.)[^:;\s]*
          captures:
            1: invalid.illegal.unexpected-character.c++
```

I think it wouldn't be wrong to even only scope the period invalid, but keep the following identifier scoped as module name. This would reduce the risk of "heavy" color changes while writing. Incomplete statements are quite common when editing.

So the following would be ok as well.

```suggestion
        - match: \.
          scope: invalid.illegal.unexpected-character.c++
```

Maybe moving it to the bottom of the context so that valid tokens are consumed first.

Reply:
Trying to prevent color changes is pretty valid.  One troublesome thing is that a module name ending in a period is invalid, so just by typing the module name you'll get invalid colors popping up every time you use a period.  Would it make sense to not flag a trailing period as invalid even though it technically is?

Reply:
That's an often discussed topic. In situations such kind of broken things may cause subtile errors it may help to see whether they are valid or not. On the other hand that's the task of a linter to highlight them. Another fact is that a the syntax definition can't highlight all kinds of illegals. So it is very likely a linter highlights invalid code, which the syntax highlights as valid.

Basically I tend to argue for trying to highlight as much as valid code as possible so we generate an impression of ST already knowing in advance what's comming next. If in doubt it might even be better to not highlight something than to highlight it invalid, especially if it is very likely while coding. This also helps mdpopups to properly highlight code fragments in GitGutters diff popup or LSP's popup.

The trailing period being highlighted invalid creates the impression as more parts are illegal. That's probably not what is expected.

We've added many illegal stuff to the JSON PR. But it turns out every here and there that it causes more issues, than being useful - especially when highlighting code fragments. On the other hand highlighting trailing commas illegal might help users new to JSON. You see .. there are arguments for and agains - we need to decide carefully.

Reply:
I've gone ahead and put a lot more work into better handling syntax highlighting while typing.  It will avoid changing colors unless you do something immediately incorrect.  Otherwise, it will catch syntax errors after the fact.